eclair-core/src/main/scala/fr/acinq/eclair/payment/receive/MultiPartHandler.scala

```diff
@@ -71,7 +71,13 @@ class MultiPartHandler(nodeParams: NodeParams, register: ActorRef, db: IncomingP
               Metrics.PaymentFailed.withTag(Tags.Direction, Tags.Directions.Received).withTag(Tags.Failure, Tags.FailureType(cmdFail)).increment()
               PendingCommandsDb.safeSend(register, nodeParams.db.pendingCommands, p.add.channelId, cmdFail)
             case None =>
-              log.info("received payment for amount={} totalAmount={}", p.add.amountMsat, p.payload.totalAmount)
+              // We log whether the sender included the payment metadata field.
+              // We always set it in our invoices to test whether senders support it.
+              // Once all incoming payments correctly set that field, we can make it mandatory.
+              p.payload.paymentMetadata match {
+                case Some(paymentMetadata) => log.info("received payment for amount={} totalAmount={} paymentMetadata={}", p.add.amountMsat, p.payload.totalAmount, paymentMetadata.toHex)
+                case None => log.info("received payment for amount={} totalAmount={} without payment metadata", p.add.amountMsat, p.payload.totalAmount)
+              }
```

Comment:
With less repetition:
```suggestion
              log.info("received payment for amount={} totalAmount={} paymentMetadata={}", p.add.amountMsat, p.payload.totalAmount, p.payload.paymentMetadata.map(_.toHex).getOrElse("none"))
```

Adding a metric would make it easier to track progress.

Reply:
I hesitated and omitted it, but since you also thought about it I'll add a metric.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2063/commits/4c39fd36e1f78f7efdfbdcc533dab6069174043f